lib/src/git.rs

```diff
@@ -1414,7 +1416,12 @@ pub fn rename_remote(
             }
         })?;
     if old_remote_name != REMOTE_NAME_FOR_LOCAL_GIT_REPO {
-        rename_remote_refs(mut_repo, old_remote_name, new_remote_name);
+        mut_repo.rename_remote(old_remote_name, new_remote_name);
+        import_some_refs(
+            mut_repo,
+            git_settings,
+            is_git_remote_ref(&[old_remote_name, new_remote_name]),
+        )?;
```

Comment:
> I am not entirely sure the semantics here are exactly the same,

There are various differences, and some of them can be disabled by config knobs (e.g. `abandon_unreachable_commits = false`, but I'm not sure if we can make `import_some_refs()` fully compatible with rename use case. Renamed remote refs shouldn't revive hidden commits, but `import_some_refs()` would do, etc.

I wouldn't try because `import_some_refs()` is already complex, and renaming/deleting refs should be simple operation.

Reply:
Right. I was annoyed at the duplication between the Git reference edits and the corresponding ones in Jujutsu’s mirror of them, but I can’t think of anything that seems to reliably simplify the logic, so I’ve just dropped that commit.